### PR TITLE
fix: [botframework-streaming] Some named pipe tests are not passing after fixing their inconclusiveness

### PR DESCRIPTION
**Pull request by erquirogasw**
Fixes # 4462
#minor

## Description
This PR fixes the NamedPipe unit tests that weren't passing.
It adds code to the _receive_ method of the _NamedPipeTransport_ class to handle null and destroyed sockets.
Additionally, it updates the transport unit test to be compatible with the new changes and the client-server tests to make the connection correctly.

## Specific Changes
  - New condition in the _receive_ method of the _NamedPipeTransport_ class to handle null and destroyed sockets.
  - Updated '_throws then reading from a dead socket_' unit test to be compatible with the changes of the _receive_ method.
  - Updated client and server '_sends without throwing_' unit tests to make the connection correctly.

## Testing
image

### CODE REVIEW

File: libraries/botframework-streaming/src/namedPipe/namedPipeTransport.ts [hunk 1]
```diff
@@ -85,7 +85,11 @@ export class NamedPipeTransport implements ITransportSender, ITransportReceiver
      * @returns The buffer containing the data from the transport.
      */
     receive(count: number): Promise<INodeBuffer> {
-        if (this._activeReceiveResolve) {
+        if (!this.socket) {
+            throw new Error('Socket unavailable to make the connection.');
+        } else if (this.socket.destroyed) {
+            throw new Error('Socket was destroyed.');
+        } else if (this._activeReceiveResolve) {
```
Comment (sw-joelmut) [inline libraries/botframework-streaming/src/namedPipe/namedPipeTransport.ts:92]: These are other error messages i think of.
throw new Error('Cannot receive data over an unavailable/null socket.');
throw new Error('Cannot receive data over a dead/destroyed socket.');

File: libraries/botframework-streaming/tests/NamedPipe.test.js [hunk 2]
```diff
@@ -166,17 +164,15 @@ describe.windowsOnly('Streaming Extensions NamedPipe Library Tests', function ()
             expect(() => transport.close()).to.not.throw();
         });
 
-        // TODO: 2023-04-24 [hawo] #4462 The code today does not allows the receive() call to be rejected by reading a dead socket.
-        //                         The receive() call will be rejected IFF the socket is closed/error AFTER the receive() call.
-        it.skip('throws when reading from a dead socket', async function () {
+        it('throws when reading from a dead socket', async function () {
```
Comment (sw-joelmut) [inline libraries/botframework-streaming/tests/NamedPipe.test.js:167]: There is a test missing where it tests from a null socket.

File: libraries/botframework-streaming/tests/NamedPipe.test.js [hunk 3]
```diff
@@ -258,20 +254,44 @@ describe.windowsOnly('Streaming Extensions NamedPipe Library Tests', function ()
...
+            class TestRequestHandler extends RequestHandler {
+                constructor() {
+                    super();
+                }
+                processRequest(_request, _logger) {
+                    return response;
+                }
+            }
+            const server = new NamedPipeServer('pipeClientTest', new TestRequestHandler());
+            const client = new NamedPipeClient('pipeClientTest', false);
```
Comment (sw-joelmut) [inline libraries/botframework-streaming/tests/NamedPipe.test.js:270]: here the false is wrong, because the second parameter is the handler.

**Review by sw-joelmut — CHANGES_REQUESTED** [review]
Review the changes i made here, maybe these is something we are missing.
joelmut/test/3-tests

File: libraries/botframework-streaming/tests/NamedPipe.test.js (hunk 3 above)
Comment (erquirogasw) [reply libraries/botframework-streaming/tests/NamedPipe.test.js:270]: Done. Thanks!

**Review by JhontSouth — APPROVED** [review]

File: libraries/botframework-streaming/tests/NamedPipe.test.js [hunk 4]
```diff
@@ -166,20 +164,31 @@ describe.windowsOnly('Streaming Extensions NamedPipe Library Tests', function ()
...
+            sock.destroyed = false;
+            sock.connecting = false;
+            sock.writable = true;
+            const transport = new NamedPipeTransport(sock, 'fakeSocket5');
+            expect(transport).to.be.instanceOf(NamedPipeTransport);
+            expect(() => transport.close()).to.not.throw();
+            expect(transport.isConnected).to.be.false;
+            expect(transport.socket).to.be.null;
+            expect(() => transport.receive(5)).to.throw();
+        });
```
Comment (sw-joelmut) [inline libraries/botframework-streaming/tests/NamedPipe.test.js:190]: Does this work?, does this reach the error 'Cannot receive data over an unavailable/null socket.'?

I remember doing something like this, where we dont pass the socket to the NamedPipeTransport.
```suggestion
        it('throws when reading from a null socket', async function () {
            const transport = new NamedPipeTransport();
            expect(transport).to.be.instanceOf(NamedPipeTransport);
            expect(transport.isConnected).to.be.false;
            expect(() => transport.receive(5)).to.throw();
            expect(() => transport.close()).to.not.throw();
        });
```

**Review by sw-joelmut — CHANGES_REQUESTED** [review]
just a small thing on the test, it could be working, but from the code i think its not?
Also, add the issue number to the pr's title

**Review by sw-joelmut — APPROVED** [review]
